compiler/rustc_lexer/src/cursor.rs

```diff
@@ -37,22 +38,19 @@ impl<'a> Cursor<'a> {
         }
     }
 
-    /// Returns nth character relative to the current cursor position.
+    /// Peeks the next symbol from the input stream without consuming it.
     /// If requested position doesn't exist, `EOF_CHAR` is returned.
     /// However, getting `EOF_CHAR` doesn't always mean actual end of file,
     /// it should be checked with `is_eof` method.
-    fn nth_char(&self, n: usize) -> char {
-        self.chars().nth(n).unwrap_or(EOF_CHAR)
-    }
-
-    /// Peeks the next symbol from the input stream without consuming it.
     pub(crate) fn first(&self) -> char {
```

Comment:
Are `first` and `second` only checked for ascii characters? In that case you could simply return the first and second byte. For non-ascii characters the first and second byte will not be valid ascii characters when interpreted individually.

Reply:
It's used in match statements all over the place, so it would awkward to change it here. But I might test that separately.

Reply:
A quick and dirty refactoring shows a 2% improvement in the microbenchmark. Not worth the complication. 
`bump()` needs to handle unicode anyway, and some helper functions operate on char.